buildSrc/src/main/java/org/elasticsearch/gradle/PublishPlugin.java

```diff
@@ -0,0 +1,202 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import com.github.jengelman.gradle.plugins.shadow.ShadowBasePlugin;
+import com.github.jengelman.gradle.plugins.shadow.ShadowExtension;
+import groovy.util.Node;
+import groovy.util.NodeList;
+import org.elasticsearch.gradle.info.BuildParams;
+import org.elasticsearch.gradle.util.Util;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.XmlProvider;
+import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.plugins.BasePluginConvention;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.tasks.GenerateMavenPom;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.bundling.Jar;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.api.tasks.javadoc.Javadoc;
+import org.gradle.external.javadoc.CoreJavadocOptions;
+import org.gradle.internal.jvm.Jvm;
+import org.gradle.language.base.plugins.LifecycleBasePlugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.elasticsearch.gradle.util.GradleUtils.maybeConfigure;
+
+public class PublishPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply("nebula.maven-base-publish");
+
+        configureJavadoc(project);
+        configureJavadocJar(project);
+        configureSourcesJar(project);
+        configurePomGeneration(project);
+
+    }
+
+    private static String getArchivesBaseName(Project project) {
+        return project.getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName();
+    }
+
+    /**Configuration generation of maven poms. */
+    private static void configurePomGeneration(Project project) {
+
+        TaskProvider<Task> generatePomTask = project.getTasks().register("generatePom");
+        // generatePomTask.configure(t -> t.dependsOn("generatePomFileForNebulaPublication"));
```

Comment:
Should this be commented out?

Reply:
Nope this can be deleted. It's handled by the last block for all publishing tasks.